### PR TITLE
Add ReplicatedMapEntryView custom type & ReplicatedMap.putAllWithMetadata [HZ-3934]

### PR DESCRIPTION
**Pull request by srknzl**
Mono pr https://github.com/hazelcast/hazelcast-mono/pull/169 

and https://github.com/hazelcast/hazelcast-mono/pull/462/

### CODE REVIEW

**Review by frant-hartm — DISMISSED** [review]

**Comment by srknzl** [comment]
@frant-hartm seems like I made your review stale can you reapprove?

File: protocol-definitions/ReplicatedMap.yaml [hunk 1]
```diff
@@ -820,3 +820,35 @@ methods:
...
             doc: |
               Number of entries affected by this event.
+  - id: 19
+    name: putAllWithMetadata
+    since: 2.7
+    doc: |
+      Copies all of the mappings from the specified entry list to this map, including metadata.
+      This method uses ReplicatedRecordStore.putRecords method with version=current version + 1.
+      Please note that all the keys in the request should belong to the partition id to which this request is being sent, all keys
+      matching to a different partition id shall be ignored. The API implementation using this request may need to send multiple
```
Comment (ihsandemir) [inline protocol-definitions/ReplicatedMap.yaml:830]: at which part of the code it is ignored? at the server side message task?

File: protocol-definitions/ReplicatedMap.yaml [hunk 2]
```diff
@@ -820,3 +820,35 @@ methods:
...
+          since: 2.7
+          doc: |
+            name of replicated map
+        - name: entries
+          type: List_ReplicatedMapEntryViewHolder
+          nullable: false
+          since: 2.7
+          doc: |
+            entries with metadata
+        - name: version
```
Comment (ihsandemir) [inline protocol-definitions/ReplicatedMap.yaml:848]: why was the version needed?

File: protocol-definitions/ReplicatedMap.yaml (hunk 1 above)
Comment (srknzl) [reply protocol-definitions/ReplicatedMap.yaml:830]: server side

File: protocol-definitions/ReplicatedMap.yaml (hunk 2 above)
Comment (srknzl) [reply protocol-definitions/ReplicatedMap.yaml:848]: it is needed by the existing API "putRecords".

**Review by frant-hartm — APPROVED** [review]

**Review by frant-hartm — DISMISSED** [review]

File: protocol-definitions/custom/Custom.yaml [hunk 3]
```diff
@@ -1448,6 +1448,38 @@ customTypes:
...
+    since: 2.7
+    returnWithFactory: false
+    params:
+      - name: key
+        type: Data
+        nullable: true
+        since: 2.7
+      - name: value
+        type: Data
+        nullable: true
```
[the remark this answers is not in the document]
Comment (srknzl) [reply protocol-definitions/custom/Custom.yaml]: fixed thx.

**Review by ihsandemir — APPROVED** [review]

**Pull request merged**